### PR TITLE
Updating to sync 2.0.0-rc23 (wrong package name)

### PR DESCRIPTION
**Pull request by kneth**
The ROS team has requested a new RC.

@blagoev @nnrealm

### CODE REVIEW

File: dependencies.list [hunk 1]
```diff
@@ -1,5 +1,5 @@
 PACKAGE_NAME=realm-js
 VERSION=2.0.0-rc9
 REALM_CORE_VERSION=3.2.1
-REALM_SYNC_VERSION=2.0.0-rc21
+REALM_SYNC_VERSION=2.0.0-rc22
 REALM_OBJECT_SERVER_VERSION=2.0.0-alpha.32
```
Comment (bmunkholm) [inline dependencies.list]: Title says rc23, you use rc22?
update to ROS to alpha.35

Comment (bmunkholm) [reply dependencies.list]: @kneth

Comment (bdash) [reply dependencies.list]: Sync's "rc23" release ended up being versioned as rc22 everywhere except the change log entry.

Comment (kneth) [reply dependencies.list]: @bmunkholm Sorry for the title being unclear. Btw, this is for a release and all sync tests are currently disabled on the `2.0.x` branch so we don't need to update ROS.

**Pull request merged**